internal/providercache/installer.go

```diff
@@ -301,6 +312,50 @@ NeedProvider:
 			continue
 		}
 
+		// if the package meta includes provider protocol versions, verify that terraform supports it.
+		if len(meta.ProtocolVersions) > 0 {
+			protoVersions, _ := versions.MeetingConstraintsStringRuby(i.pluginProtocolVersion)
```

Comment:
I think with the string constant you set above this would not accept versions `5.1` or `5.2`, right? At the moment that's kinda moot because we never updated the release script to send anything other than `5.0`, but maybe we could keep our options open for using that mechanism for minor version selection later by setting this string to something like `~> 5`, or `>= 5, < 6`?

(Technically we shouldn't need to constrain by minor version because the minor version updates are supposed to be feature-sniffable and backwards-compatible, so this is mainly motivated by giving us some flexibility in technical solutions to later problems we've not anticipated yet.)

Reply:
AHHHH thank you! That's what I was missing when reading the original code in plugin/discovery. I understand much more now 😀